dev-lang/zig/zig-0.7.1.ebuild

```diff
@@ -0,0 +1,52 @@
+# Copyright 2019-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake llvm
+
+DESCRIPTION="A robust, optimal, and maintainable programming language"
+HOMEPAGE="https://ziglang.org/"
+LICENSE="MIT"
+SLOT="0"
+IUSE="+experimental"
+
+if [[ ${PV} == 9999 ]]; then
+	EGIT_REPO_URI="https://github.com/ziglang/zig.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/ziglang/zig/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+# According to zig's author, zig builds that do not support all targets are not
+# supported by the upstream project.
+ALL_LLVM_TARGETS=(
+	AArch64 AMDGPU ARM AVR BPF Hexagon Lanai Mips MSP430 NVPTX
+	PowerPC RISCV Sparc SystemZ WebAssembly X86 XCore
+)
+ALL_LLVM_TARGETS=( "${ALL_LLVM_TARGETS[@]/#/llvm_targets_}" )
+LLVM_TARGET_USEDEPS="${ALL_LLVM_TARGETS[@]}"
+
+LLVM_MAX_SLOT=11
+
+RDEPEND="
+	sys-devel/clang:${LLVM_MAX_SLOT}
+	>=sys-devel/lld-11.0.0
+	<sys-devel/lld-12.0.0
```

Comment:
Not to be super annoying, but this doesn't actually work. A valid package (and the one resolving for me!) is `sys-devel/lld-11.0.1`, which hasn't been touched my this PR. Would a solution be to just backport the fix to existing ebuilds? unmasking the new revisions? adding a `shared` flag to existing ebuilds? I can file something if you give me a good idea and you want @mgorny

Reply:
See: https://bugs.gentoo.org/778398